### DISCUSSION TITLE
Is MaxToolbar still supported?

### DISCUSSION THREAD

**Discussion by LTolDe** [maxGraph/maxGraph#251] [Q&A]
The function I passed to the MaxToolbar.addMode() method does not get executed on mouse click.
My main script looks like this:
```
const graphContainer = <HTMLElement>document.getElementById('graph-container');
const graph = new Graph(graphContainer);
const parent = graph.getDefaultParent();
graph.batchUpdate(() => {
...
});
const toolbarContainer = <HTMLElement>document.getElementById('toolbar-container');
const toolbar = new MaxToolbar(toolbarContainer);
toolbar.addMode(
  'test',
  iconUrl,
  (_evt: MouseEvent, cell: Cell) => {
    console.log('my event handler');
  },
  iconUrl
)
```
The toolbar examples in the stories
[Toolbar](https://github.com/maxGraph/maxGraph/blob/v0.4.0/packages/html/stories/Toolbar.stories.js#L132-L144)
[Dynamic Toolbar](https://github.com/maxGraph/maxGraph/blob/v0.4.0/packages/html/stories/DynamicToolbar.stories.js#L120-L138)
and the old mxGraph javascript examples
[Toolbar](https://jgraph.github.io/mxgraph/javascript/examples/toolbar.html)
[Dynamic Toolbar](https://jgraph.github.io/mxgraph/javascript/examples/dynamictoolbar.html)
all use gestureUtils.makeDraggable() (or mxUtils.makeDraggable()). However, I do not wish to make the toolbar item draggable. I want the click events on the graph to be handled differently according to the selected mode in the toolbar.

**Comment by tbouffard** [comment]
Hi, MaxToolbar is not deprecated.

Can you share some code please?
We need more context to better understand your problem.

**Reply by LTolDe** [reply]
Thank you for your response. I have added some code to the question.

**Reply by tbouffard** [reply]
OK thanks for sharing the code.
AFAIK, this method is working. It is used in the Stories and everything works fine.
- https://github.com/maxGraph/maxGraph/blob/v0.4.0/packages/html/stories/Toolbar.stories.js#L132-L144
- https://github.com/maxGraph/maxGraph/blob/v0.4.0/packages/html/stories/DynamicToolbar.stories.js#L120-L138

I recently fixed a problem with `Toolbar.stories `in #230, but it is not related to the usage of `addMode`.

**Reply by LTolDe** [reply]
Both stories call gestureUtils.makeDraggable(). I do not wish to make the toolbar item draggable. I want the click events on the graph to be handled differently according to the selected mode in the toolbar. I thought MaxToolbar.addMode() was meant to offer this functionality. Is it actually meant to only be used with gestureUtils.makeDraggable()?

**Reply by tbouffard** [reply]
I don't know how `addMode` behaves or is supposed to behave. So I'll let other people answer your question.

Have you checked how this was done in the mxGraph examples to see how it used to be?
- https://jgraph.github.io/mxgraph/javascript/examples/toolbar.html
- https://jgraph.github.io/mxgraph/javascript/examples/dynamictoolbar.html

**Reply by LTolDe** [reply]
Thank you for your responses. Both mxGraph examples use mxUtils.makeDraggable(), which does not meet my needs.
